### DISCUSSION TITLE
Allow to link issue&PR directly by URL

### DISCUSSION THREAD

**Discussion by mickaelistria** [community/community#118423] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Issues

### Body

I like to link issues and PRs a lot on GitHub.com.
However, this often requires more clicks/focus/human memory than necessary as in the most cases, I would like to paste the direct URL of the target issue. The current UI doesn't allow any of those simple cases to work and requires to navigate first to project and then to pick an issue/PR. Then the workflow would become: copy the URL of the target issue, move to the source one, click on "Development", Ctrl+V, Enter, and done; no project name nor issue # to find or remember, less risk of errors (assuming the right link was copied first).

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by jfullstackdev** [comment]
here are some of the suggestion from Copilot

---

GitHub provides several features that can help you link issues and PRs more efficiently:

1. **Autocomplete**: When you're typing in a comment, you can type `#` followed by the start of the issue or PR number or title, and GitHub will show an autocomplete dropdown with matching issues and PRs. This can save you from having to remember or look up the exact number.

2. **Keyboard Shortcuts**: GitHub supports a variety of keyboard shortcuts that can help you navigate more quickly. For example, you can press `g` then `i` to go to your issues, or `g` then `p` to go to your pull requests.

3. **Saved Replies**: If you find yourself typing the same thing over and over again, you can create a saved reply. This can be useful for linking to common issues or PRs.

4. **Browser Extensions**: There are several browser extensions that can enhance GitHub's functionality. For example, [OctoLinker](https://octolinker.now.sh/) can turn issue and PR numbers into clickable links.

5. **GitHub CLI**: If you're comfortable with the command line, you can use the [GitHub CLI](https://cli.github.com/) to manage issues and PRs. This can be faster than using the web interface, especially for complex tasks.

Remember, the best tool or method depends on your specific workflow and preferences. It might take some experimentation to find what works best for you.

**Reply by mickaelistria** [reply]
Thanks, I know about almost all of them; but I still think that allowing me to paste an URL would be in my case a much simpler, more intuitive, more efficient, more portable, more universal... approach than any of those.

**Comment by KittyChiu** [comment]
If you want to...

1. **Associate** an issue with PR

One of my favourite key is `#`. By pressing `#` within an issue or issue comment and some keywords or issue numbers, it brings an IntelliSense dropdown box like this one for you to select and **associate** an issue with PR.

![image](https://github.com/community/community/assets/42864823/c96e337b-bf2f-4f61-9353-abadfe74797f)
 
2. **Link** an issue with PR

You can use one of [these keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) with the issue number like `fix #123` inside PR or PR comment, and the PR will be tracked by the issue.

**Reply by mickaelistria** [reply]
Thanks, I'm already aware of that. But I still have to first select a repo and to remember a # or a keyword to achieve it; while I have a link in my buffer.
My comment is more a request for enhancement targeting GitHub than a request for user support.
